### PR TITLE
Add clusterProfile docs

### PR DESCRIPTION
**Pull request by zhiying-lin**
Add ClusterProfile project links to the Contributing section and high level documentation to the Overview section.

Fix: #19

### CODE REVIEW

File: site-src/concepts/cluster-profile-api.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# ClusterProfile API Overview
+
+This document provides an overview of the ClusterProfile (fka ClusterInventory) API.
```
Comment (mikeshng) [inline site-src/concepts/cluster-profile-api.md:3]:
```suggestion
This document provides an overview of the ClusterProfile API.
```

We don't need to mention the formally know as part here. ClusterInventory API on a conceptual level still exists and the sig repo is cluster-inventory-api.

**Review by mikeshng — APPROVED** [review]
/lgtm

File: site-src/concepts/cluster-profile-api.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# ClusterProfile API Overview
+
+This document provides an overview of the [ClusterProfile API](https://github.com/kubernetes-sigs/cluster-inventory-api?tab=readme-ov-file#cluster-profile-api).
+
+![Alt](../images/cluster-profile-api.png "ClusterProfile API")
+
+A Cluster Profile is essentially an individual member of the Cluster Inventory that details the properties and status of a cluster.
```
Comment (corentone) [inline site-src/concepts/cluster-profile-api.md:7]:
```suggestion
A Cluster Profile is essentially an individual member of the Cluster Inventory that details properties and status of a cluster.

```

File: site-src/concepts/cluster-profile-api.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# ClusterProfile API Overview
+
+This document provides an overview of the [ClusterProfile API](https://github.com/kubernetes-sigs/cluster-inventory-api?tab=readme-ov-file#cluster-profile-api).
+
+![Alt](../images/cluster-profile-api.png "ClusterProfile API")
+
+A Cluster Profile is essentially an individual member of the Cluster Inventory that details the properties and status of a cluster.
+This API proposes a universal, standardized interface that defines how cluster information should be presented
```
Comment (corentone) [inline site-src/concepts/cluster-profile-api.md:8]:
```suggestion
This API proposes a standardized interface that defines how cluster information should be presented

```

File: site-src/concepts/cluster-profile-api.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  and keeps their status up-to-date. Each cluster manager MUST be identified with a unique name.  
+  Each ClusterProfile resource SHOULD be managed by only one cluster manager. A cluster manager SHOULD
+  have sufficient permission to access the member cluster to fetch the information so it can update the status
+  of the ClusterProfile API resource.
+
+- **ClusterProfile API Consumer**: the person running the cluster managers
+  or the person developing extensions for cluster managers for the purpose of
+  workload distribution, operation management etc.
+
+## API Example
```
Comment (corentone) [inline site-src/concepts/cluster-profile-api.md:27]: Could we point to the CRD definition? I think it's in sig-mc too.
@mikeshng would know the location.

File: site-src/concepts/cluster-profile-api.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+- **Cluster Manager**: An entity that creates the ClusterProfile API object per member cluster,
+  and keeps their status up-to-date. Each cluster manager MUST be identified with a unique name.  
+  Each ClusterProfile resource SHOULD be managed by only one cluster manager. A cluster manager SHOULD
+  have sufficient permission to access the member cluster to fetch the information so it can update the status
+  of the ClusterProfile API resource.
+
+- **ClusterProfile API Consumer**: the person running the cluster managers
+  or the person developing extensions for cluster managers for the purpose of
+  workload distribution, operation management etc.
+
```
Comment (corentone) [inline site-src/concepts/cluster-profile-api.md:26]: Could you add one sentence to add the following key information: 1/ the relationship to namespace, 2/ the relationship to clusterSet.

something like:
```
Cluster Profiles are namespace-level resources and ClusterProfiles of the same namespace MUST be in the same clusterSet or no clusterSet at all.
```

File: site-src/images/cluster-profile-api.png
Comment (corentone) [inline site-src/images/cluster-profile-api.png:1]: Could we represent the actual clusters in this picture?

Show that clusterProfile is really a "pointer" to a real cluster.
The inventory doesn't own the clusters themselves, it owns pointers to them.

(Though I guess technically some could argue that the ClusterProfile could be used as the source object that then creates a cluster, this is not the original design for it afaict)

**Review by corentone — COMMENTED** [review]
thank you for writing this page. There is a delicate balance between putting some on the page and just dumping the KEP on the page.

File: site-src/concepts/cluster-profile-api.md (hunk 5 above)
Comment (zhiying-lin) [reply site-src/concepts/cluster-profile-api.md:28]: included them in the text.

**Review by ryanzhang-oss — APPROVED** [review]

**Review by mikeshng — APPROVED** [review]
/lgtm

File: site-src/concepts/cluster-profile-api.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+## API Example
+
+[CRD definition](https://github.com/kubernetes-sigs/cluster-inventory-api/blob/main/config/crd/bases/multicluster.x-k8s.io_clusterprofiles.yaml)
+
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ClusterProfile
+metadata:
+ name: generated-cluster-name
```
Comment (corentone) [inline site-src/concepts/cluster-profile-api.md:37]: I think this is missing a namespace.
Can you confirm this example applies?

Comment (zhiying-lin) [reply site-src/concepts/cluster-profile-api.md]: good catch! resolved

**Review by corentone — APPROVED** [review]

**Comment by JeremyOT** [comment]
Thanks for doing this, looks good

/lgtm
/approve

**Pull request merged**